lint/linter.py

```diff
@@ -15,6 +15,20 @@
 NEAR_RE_TEMPLATE = r'(?<!"){}({}){}(?!")'
 BASE_CLASSES = ('PythonLinter',)
 
+# Many linters use stdin, and we convert text to utf-8
+# before sending to stdin, so we have to make sure stdin
+# in the target executable is looking for utf-8. Some
+# linters (like ruby) need to have LANG and/or LC_CTYPE
+# set as well.
+UTF8_ENV_VARS = {
+    'PYTHONIOENCODING': 'utf8',
+    'LANG': 'en_US.UTF-8',
+    'LC_CTYPE': 'en_US.UTF-8',
+}
+
+BASE_LINT_ENVIRONMENT = ChainMap(UTF8_ENV_VARS, os.environ)
```

Comment:
Generally, this is nice to read code.

In this version, the environment during **runtime** would be `os.environ` plus a few *utf8* settings. Of course, we could also augment our PATH here. I just don't know if that's necessary. 

We usually use the augmented path for our `which` calls. And this is used if we actually have a real executable. *The runtime environment.*